src/OauthTokenFileStorage.php

```diff
@@ -0,0 +1,179 @@
+<?php
+
+/**
+ * Copyright 2018 Google Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License version 2 as published by the
+ * Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public
+ * License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc., 51
+ * Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+
+namespace Drupal\apigee_edge;
+
+use Apigee\Edge\HttpClient\Plugin\Authentication\OauthTokenStorageInterface;
+
+/**
+ * Storing and returning OAuth access token data.
+ */
+class OauthTokenFileStorage implements OauthTokenStorageInterface {
+
+  /**
+   * The storage location for oauth data.
+   */
+  const OAUTH_TOKEN_PATH = 'private://.apigee_edge/oauth.dat';
```

Comment:
Should not consistent in the path and the name of the files that we usage as a storage? We have `private://apigee_edge...` something above and we have this here (which kinda looks like it would have come from the Drupal 7 version).

Reply:
Also, should not we allow to change to location and the file name with a config in this storage at least? (Similarly how it worked in Drupal 7). If we use this _unencrypted_ storage by default, we should allow users to move this file to wherever they want on their server that they feel even more secure than the private file system.
Like the nobackup folder on ACQ: https://docs.acquia.com/acquia-cloud/manage/files/system-files/private/

@cnovak ^

Reply:
Make configurable via `settings.php`